### PR TITLE
Replace reflection code with actual method from Accumulo

### PR DESCRIPTION
**Pull request by milleruntime**
* Replace reflection code that was making multiple guesses at an
internal Accumulo method with a copy of the actual method

The IteratorUtil method was moved in Accumulo 2.0.  The one here I copied from 1.9
https://github.com/apache/accumulo/blob/1.9/core/src/main/java/org/apache/accumulo/core/iterators/IteratorUtil.java#L124

### CODE REVIEW

File: warehouse/ingest-core/src/main/java/datawave/ingest/table/config/AbstractTableConfigHelper.java [hunk 1]
```diff
@@ -97,33 +97,18 @@ protected void setAggregatorConfigurationIfNecessary(String tableName, List<Comb
     }
     
     /**
-     * This calls the generateInitialTableProperties on the IteratorUtil class using reflection because they changed the signature between 1.4.1 and 1.4.2 of
-     * ACCUMULO and did not supply any backward compatibility (argh).
-     * 
-     * @return
+     * Copied from Accumulo 1.9 IteratorUtil
```
Comment (ivakegg) [inline warehouse/ingest-core/src/main/java/datawave/ingest/table/config/AbstractTableConfigHelper.java]: Is this implementation valid for 2.0?

Comment (milleruntime) [reply warehouse/ingest-core/src/main/java/datawave/ingest/table/config/AbstractTableConfigHelper.java]: Yeah the code was just moved again.

**Comment by ivakegg** [comment]
Can you rebase against master please

**Comment by milleruntime** [comment]
> Can you rebase against master please

Done.

**Comment by brianloss** [comment]
> 
> 
> > Can you rebase against master please
> 
> Done.

You might need to rebase one more time -- CI is failing due to a missing artifact for the quickstart tests. I believe I have fixed that and just pushed to master.

**Comment by brianloss** [comment]
Although the failure in the unit tests is not expected.

**Comment by milleruntime** [comment]
> > > Can you rebase against master please
> > 
> > 
> > Done.
> 
> You might need to rebase one more time -- CI is failing due to a missing artifact for the quickstart tests. I believe I have fixed that and just pushed to master.

Rebased again.

**Comment by milleruntime** [comment]
> Although the failure in the unit tests is not expected.

I see the failures now.  It looks like it can't find the VersioningIterator.  This must be related to other recent changes.

**Comment by keith-ratcliffe** [comment]
> You might need to rebase one more time -- CI is failing due to a missing artifact for the quickstart tests. I believe I have fixed that and just pushed to master.

@brianloss, I noticed in the the quickstart test log that ingest jobs are failing here with the CNFE below...
```
Exception in thread "main" [0m[91mjava.lang.ClassNotFoundException: VersioningIterator
2019-10-22T17:03:15.3043485Z [[0m[91m	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
2019-10-22T17:03:15.3043870Z 	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
2019-10-22T17:03:15.3044141Z 	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
2019-10-22T17:03:15.3044362Z 	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
2019-10-22T17:03:15.3044548Z 	at java.lang.Class.forName0(Native Method)
2019-10-22T17:03:15.3044747Z 	at java.lang.Class.forName(Class.java:264)
2019-10-22T17:03:15.3044963Z 	at datawave.ingest.mapreduce.job.IngestJob.serializeAggregatorConfiguration(IngestJob.java:1304)
2019-10-22T17:03:15.3045354Z 	at datawave.ingest.mapreduce.job.IngestJob.run(IngestJob.java:308)
2019-10-22T17:03:15.3045585Z 	at org.apache.hadoop.util.ToolRunner.run(ToolRunner.java:76)
2019-10-22T17:03:15.3045802Z 	at datawave.ingest.mapreduce.job.IngestJob.main(IngestJob.java:210)
2019-10-22T17:03:15.3046027Z 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
```
....and then I recalled encountering the same issue while setting up the hadoop3 pipeline in gitlab.

The way I got around this issue there was to stage a pre-built, hadoop3-specific Accumulo tarball on a uniquely tagged gl-runner. 

Unfortunately, there's no downloadable distro of 1.9.3 that is built with Accumulo's hadoop 3 Maven profile (which resolves the discrepancy in dependencies surfaced above), so I had to roll my own. Perhaps we could address that in the same fashion that you used for staging the local maven repo(?)

**Comment by brianloss** [comment]
That's pretty unfortunate that Accumulo has those profiles but doesn't distribute any release with them enabled. If we have some publicly available place to put such as distro, then that would be best -- just download it. Otherwise, yeah we have to do a big hack, but the maven image trick isn't the right long-term answer.

**Comment by brianloss** [comment]
@keith-ratcliffe one other thing to consider--the build runs on release/version3.0 and master. See [here](https://github.com/NationalSecurityAgency/datawave/commit/a5490aed18cd2871bfdf84cbe10190fc7e74285e/checks?check_suite_id=276151971) and [here](https://github.com/NationalSecurityAgency/datawave/commit/dc6e1c731d182147886874a3aab3c0e4887019ba/checks?check_suite_id=276150610). So either that build has some kind of race condition where it passes sometimes and will fail with this error others, or it's due to the changes in this PR, though I'm not sure how the latter would be causing it.

**Comment by keith-ratcliffe** [comment]
> ...So either that build has some kind of race condition where it passes sometimes and will fail with this error others, or it's due to the changes in this PR, though I'm not sure how the latter would be causing it.

Yeah, I'm confused. Although, I suppose the recent cherry-pick to make release/version3.0 could have missed something that would account for difference in behavior as well. Dunno yet. I'll investigate a little more

File: warehouse/ingest-core/src/main/java/datawave/ingest/table/config/AbstractTableConfigHelper.java [hunk 2]
```diff
@@ -97,33 +97,18 @@ protected void setAggregatorConfigurationIfNecessary(String tableName, List<Comb
...
-                props = (Map<String,String>) (method.invoke(null, true));
-            } catch (Exception ex) {
-                throw new RuntimeException("Unable to call generateInitialTableProperties", ex);
-            }
-        } catch (Exception e) {
-            throw new RuntimeException("Unable to call IteratorUtil.generateInitialTableProperties()", e);
+        TreeMap<String,String> props = new TreeMap<>();
+        
+        for (IteratorScope iterScope : IteratorScope.values()) {
+            props.put(Property.TABLE_ITERATOR_PREFIX + iterScope.name() + ".vers", "20,VersioningIterator");
```
Comment (brianloss) [inline warehouse/ingest-core/src/main/java/datawave/ingest/table/config/AbstractTableConfigHelper.java]: `VersioningIterator` doesn't have a package name. I'm guessing maybe that's leading to the test errors.

File: warehouse/ingest-core/src/main/java/datawave/ingest/table/config/AbstractTableConfigHelper.java [hunk 3]
```diff
@@ -97,33 +97,18 @@ protected void setAggregatorConfigurationIfNecessary(String tableName, List<Comb
...
+            props.put(Property.TABLE_ITERATOR_PREFIX + iterScope.name() + ".vers.opt.maxVersions", "1");
         }
+        
+        props.put(Property.TABLE_CONSTRAINT_PREFIX.toString() + "1", "DefaultKeySizeConstraint");
```
Comment (brianloss) [inline warehouse/ingest-core/src/main/java/datawave/ingest/table/config/AbstractTableConfigHelper.java]: Same problem here with a missing package name.

**Pull request merged**